### PR TITLE
Reference implementation license

### PR DESCRIPTION
**Pull request by protolambda**
Use a MIT license for the reference implementation.

The specification is still CC0 1.0 Universal, but we should use a software license for the reference implementation.

The MIT license specifically was chosen by @karlfloersch, since it follows the same licensing as other previous Optimism works (many parts of the existing [monorepo](https://github.com/ethereum-optimism/optimism))

More information about the license: https://choosealicense.com/licenses/mit/

### CODE REVIEW

**Review by karlfloersch — APPROVED** [review]
LGTM!

**Comment by karlfloersch** [comment]
Don't forget to FOSS

File: README.md [hunk 1]
```diff
@@ -45,4 +45,6 @@ shared publicly (we will ask for consent before recording).
 
 ## License
 
-CC0 1.0 Universal, see [`LICENSE`](./LICENSE) file.
+Specification: CC0 1.0 Universal, see [`LICENSE`](./LICENSE) file.
+
+Reference implementation: MIT, see [`opnode/LICENSE`](./opnode/LICENSE) file.
```
Comment (maurelian) [inline README.md:50]: What about contracts?

Comment (protolambda) [reply README.md]: Good clarification point! Smart contract sources have been licensed as `MIT` in the main optimism monorepo, and should be thought of as part of the reference software, so these should also be `MIT` licensed.

**Comment by protolambda** [comment]
Changes:
- Move specification license (CC0 1.0 Universal) to `specs` directory (not long ago the specification docs moved to there as well)
- Clarify reference smart contracts license with explicit MIT `LICENSE` file in `packages/contracts`
- Keep MIT `LICENSE` in `opnode` directory to be explicit about the reference implementation of the rollup-node
- Clarify licensing in readme
- `Reference software` note + `LICENSE` in root directory: covers everything not in the `specs` directory

**Review by karlfloersch — APPROVED** [review]
🚢

**Pull request merged**